### PR TITLE
Enhance float number comparison

### PR DESCRIPTION
**Pull request by luckyxiaoqiang**
### Describe what this PR does / why we need it
Enhance float number comparison.

### Does this pull request fix one issue?
#281 

### Special notes for reviews

### CODE REVIEW

File: adapter/grpc/server_test.go [hunk 1]
```diff
@@ -6,12 +6,14 @@ import (
 	"os"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc"
+
```
Comment (louyuting) [inline adapter/grpc/server_test.go:11]: refine imports by "go imports"

File: adapter/micro/client_test.go [hunk 2]
```diff
@@ -6,15 +6,17 @@ import (
 	"log"
 	"testing"
 
+	"github.com/micro/go-micro/v2/client"
+	"github.com/micro/go-micro/v2/client/selector"
+	"github.com/micro/go-micro/v2/registry/memory"
+	"github.com/stretchr/testify/assert"
+
```
Comment (louyuting) [inline adapter/micro/client_test.go:13]: refine imports by "go imports"

File: adapter/micro/server_test.go [hunk 3]
```diff
@@ -7,14 +7,16 @@ import (
 	"testing"
 	"time"
 
+	"github.com/micro/go-micro/v2"
+	"github.com/micro/go-micro/v2/server"
+	"github.com/stretchr/testify/assert"
+
```
Comment (louyuting) [inline adapter/micro/server_test.go:13]: refine imports by "go imports"

File: adapter/grpc/server_test.go (hunk 1 above)
Comment (luckyxiaoqiang) [reply adapter/grpc/server_test.go]: Addressed.

File: adapter/micro/client_test.go (hunk 2 above)
Comment (luckyxiaoqiang) [reply adapter/micro/client_test.go]: Addressed.

File: adapter/micro/server_test.go (hunk 3 above)
Comment (luckyxiaoqiang) [reply adapter/micro/server_test.go]: Addressed.

**Review by louyuting — APPROVED** [review]
Thanks for contribution

**Pull request merged**